hail/Makefile

```diff
@@ -87,16 +79,16 @@ python/hailctl/deploy.yaml:
 	echo "dataproc:" >> python/hailctl/deploy.yaml
 	for FILE in `ls python/hailctl/dataproc/resources`; do \
 	  CLOUD_LOC=$(HAILCTL_BUCKET_BASE)/$(DEV_CLARIFIER)/$(CLOUD_SUB_FOLDER)/$$FILE && \
-	  gsutil -m cp python/hailctl/dataproc/resources/$$FILE $$CLOUD_LOC && \
-	  gsutil acl set public-read $$CLOUD_LOC && \
 	  echo "  $$FILE: $$CLOUD_LOC" >> python/hailctl/deploy.yaml || break; done || exit
-	echo "  jar: $(jar_cloud_path)" >> python/hailctl/deploy.yaml
 	echo "  wheel: $(wheel_cloud_path)" >> python/hailctl/deploy.yaml
+	echo "  pip_dependencies: $(shell cat python/requirements.txt | grep -v pyspark | tr "\n" "|||")" >> python/hailctl/deploy.yaml
 
-.PHONY: upload-deferred-artifacts
-upload-deferred-artifacts: wheel
-	gsutil -m cp build/libs/hail-all-spark.jar $(jar_cloud_path)
-	gsutil acl set public-read $(jar_cloud_path)
+.PHONY: upload-artifacts
+upload-artifacts: wheel
+	for FILE in `ls python/hailctl/dataproc/resources`; do \
```

Comment:
Why not make this an explicit listing of the files in that directory? We can then use the `notdir` and `addprefix` makefile functions to reduce the number of times gsutil is invoked.

What I'm thinking of is something like.
```make
        # not in this rule
	RESOURCES = $(wildcard python/hailctl/dataproc/*)
	CLOUD_RESOURCES = $(addprefix $(cloud_base)/, $(notdir $(RESOURCES)))
        # inside this rule
	gsutil -m cp $(wheel_path) $(RESOURCES) $(cloud_base)
	gsutil -m acl set public-read $(wheel_cloud_path) $(CLOUD_RESOURCES)
```

Reply:
oh, nice.

Reply:
I don't entirely know what the wildcard/notdir/addprefix stuff does -- I fixed it to use an explicit shell wildcard in the rule, which seems to work fine.